libraries/system/system_calls.h

```diff
@@ -223,6 +223,7 @@ static inline return_type system_call_thread_create(thread_entry_point_type *thr
 
     asm volatile("pushl %1\n"
                  "pushl %2\n"
+                 "pushl %%esp\n" // Need to be last (i.e. first) because %1 and %2 will be %esp-relative references.
```

Comment:
Obviously, this is silly since _the ESP value is already pushed on the stack_ (by the CPU), when making a far call to a segment with another privilege level (see _Intel Architecture Software Developer’s Manual:  Instruction Set
Reference, 3-55_). But, it's much more awkward from a C function to read that value (since we are compiling with `-fomit-frame-pointer`; otherwise it would have been reasonable) so we chose the simple way for now.